torcharrow/idataframe.py

```diff
@@ -56,6 +56,14 @@ def DataFrame(
         where possible.  Should be a dt.Struct() providing a list of
         dt.Fields.
 
+    columns: list of strings, default None
```

Comment:
list of strings is good for now. Note in the future we might want to support nested column names. But how to specify it is still open. We can use a dict but it looks yunky for non-struct type: 
```python
ta.DataFrame(
    [(1, 2, (3, 4))], 
    columns={
           "a": None, 
           "b": None, 
           "nest": ["n1", "n2"]
   }
)
```

Or we can have a recursive type `FieldDesc`:
```python
class FieldDesc:
    fieldName: str
    nested_fields: Optional[List[Union[String, FieldDesc]]]
```

```python
ta.DataFrame(
    [(1, 2, (3, 4))], 
    columns=["a", "b", FieldDesc("nest", ["n1", "n2"])]
)
```

Reply:
Does allowing only NamedTuple (no "pure" tuple) sound like an option?